### PR TITLE
[New Builder Suite] - Date & Time Suite 

### PR DESCRIPTION
**Pull request by huntabyte**
This PR introduces an entire "suite" of new builders - **Date & Time Suite** which consists of the following builders:

- **Calendar** - build customizable, accessible, and powerful calendar components
- **Date Field** - an enhanced alternative to the native `<input type="date" />` with an emphasis on localization & accessibility
- **Date Picker** - a combination of the _Calendar_, _Date Field_, &  _Popover_ builders
- **Range Calendar** - build a calendar with range selection. Extends the power & functionality of the _Calendar_
- **Date Range Field** - an enhanced alternative to using two native `<input type="date" />` inputs for selecting start and end dates, with an emphasis on localization & accessibility 
- **Date Range Picker** - a combination of the _Range Calendar_, _Date Range Field_, & _Popover_ builders

Please note, this is a large introduction into the library, and although we have tests, we can't possibly catch all the bugs, so if you find any, please create an issue and we'll address it.

Additionally, if anyone would like to contribute to the documentation, we always welcome assistance there!

### Todos:
- [x] Cleanup/finalize docs
- [x] Mobile calendar styles
- [x] API References 😭
- [x] Make tests pass regardless of the GH runner's timezone
- [x] Add events/ActionReturn types
- [x] BYOID

### CODE REVIEW

**Comment by huntabyte** [comment]
Should be good to start review @TGlide, the remainder of the due outs are more style-related.

**Comment by theetherGit** [comment]
@huntabyte how much time will it take to release this builder one?? I'm looking for Date and Time component.

**Comment by TGlide** [comment]
> @huntabyte how much time will it take to release this builder one?? I'm looking for Date and Time component.

F8Ppo8iaAAIONzu

File: src/docs/content/builders/date-field.md [hunk 1]
```diff
@@ -0,0 +1,548 @@
...
+	<!-- ...rest -->
+</div>
+```
+
+It's not recommended, as the formatting doesn't adapt to the locale and type of date being
+represented, which is one of the more powerful features this builder provides.
+
+Instead, you can use the `segmentContents` state, which is an array of objects necessary to form the
+date. Each object has a <Code>part</Code> property, which is the `SegmentPart`, and a
+<Code>value</Code> property, which is the locale-aware string representation of the segment.
```
Comment (TGlide) [inline src/docs/content/builders/date-field.md:103]: This resulted in a weird line break

File: .eslintrc.cjs [hunk 2]
```diff
@@ -45,7 +45,7 @@ module.exports = {
 		// eslint-plugin-svelte
 		'svelte/no-target-blank': 'error',
 		'svelte/no-immutable-reactive-statements': 'error',
-		'svelte/no-trailing-spaces': 'error',
+		'svelte/no-trailing-spaces': 'off', // I HATE THIS RULE
```
[the remark this answers is not in the document]
Comment (huntabyte) [reply .eslintrc.cjs]: It's actually removed altogether in our develop branch now 😂 basically as soon as you open a function it instantly gives you linting errors about white space before you can even format.

File: src/docs/content/builders/date-field.md (hunk 1 above)
Comment (huntabyte) [reply src/docs/content/builders/date-field.md]: ✅

**Pull request merged**